pages/box_buster.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <b-container>
+    <h1>Box Buster</h1>
+    <p class="lead">
+      Quickly overview box contents to assist with cherrypicking
+    </p>
+    <b-form-group
+      id="box-barcode"
+      description="Scan in a lighthouse box barcode to show information about all plates contained in the box."
+      label="Please scan lighthouse box barcode"
+      label-for="box-barcode-field"
+      :invalid-feedback="labwhereFeedback"
+      valid-feedback="Box found"
+      :state="labwhereState"
+    >
+      <b-form-input
+        id="box-barcode-field"
+        v-model="barcode"
+        type="search"
+        trim
+        :state="labwhereState"
+        @change="findBox"
+      >
+      </b-form-input>
+    </b-form-group>
+    <b-table
+      v-if="showTable"
+      :items="plates"
+      :tbody-tr-class="rowClass"
+      show-empty
+      :empty-text="lighthouseFeedback"
+      sort-by="number_of_positives"
+      :sort-desc="true"
+      :fields="fields"
+    >
+      <template #table-caption>
+        <span>Box Summary:</span>
+        <span>Total of {{ total }} plates in box;</span>
+        <span>{{ total_with_maps }} plates with plates maps,</span>
+        <span>{{ total_without_maps }} without.</span>
+        <span>Total {{ total_positives }} positives.</span>
+      </template>
+    </b-table>
+  </b-container>
+</template>
+
+<script>
+import lighthouse from '../modules/lighthouse_service'
+import { getPlatesFromBoxBarcodes } from '@/modules/labwhere'
+
+// import lighthouse from '../modules/lighthouse_service'
+const countWithMap = (accumulator, plate) =>
```

Comment:
This is all good but wonder if it would sit better in a module especially as it is outside the actual component. I know they are simple but are there any tests for it? I just wasn't clear on exactly what is going on.

Reply:
These are all tested as part of the behaviour of the component itself, following the vue-test-utils guidelines of:

> You want to write maintainable tests for your Vue components. As a part of this goal, you want your tests to avoid including implementation details of your components. You'd rather focus on making your tests give you the confidence for which they are intended.

Obviously this wouldn't apply any more if I pulled them into a module, as they'd be exposed independently, so I'd add tests then.